tests/pausing_test.go

```diff
@@ -412,28 +406,43 @@ var _ = Describe("[rfe_id:3064][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			Expect(expecterErr).ToNot(HaveOccurred())
 			defer expecter.Close()
 
-			By("Starting a process")
-			startPid := startProcess(expecter)
-			Expect(startPid).ToNot(BeEmpty())
+			By("checking uptime difference between guest and host")
+			uptimeDiffBeforePausing = hostUptime() - grepGuestUptime(expecter)
+		})
+
+		AfterEach(func() {
+			if vmi != nil {
+				By("Delete VMI")
+				Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &v13.DeleteOptions{})).To(Succeed())
 
+				Eventually(func() error {
+					_, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &v13.GetOptions{})
+					return err
+				}, time.Minute, time.Second).Should(
+					SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())),
+					"The VMI should be gone within the given timeout",
+				)
+			}
+		})
+
+		It("[test_id:3090]should be less than uptime difference after pause", func() {
 			By("Pausing the VMI")
 			command := tests.NewRepeatableVirtctlCommand("pause", "vmi", "--namespace", tests.NamespaceTestDefault, vmi.Name)
 			Expect(command()).To(Succeed())
 			tests.WaitForVMICondition(virtClient, vmi, v1.VirtualMachineInstancePaused, 30)
 
-			By("Waiting longer than the process normally runs")
-			time.Sleep(10 * time.Second)
-
 			By("Unpausing the VMI")
 			command = tests.NewRepeatableVirtctlCommand("unpause", "vmi", "--namespace", tests.NamespaceTestDefault, vmi.Name)
 			Expect(command()).To(Succeed())
 			tests.WaitForVMIConditionRemovedOrFalse(virtClient, vmi, v1.VirtualMachineInstancePaused, 30)
 
-			By("Checking the process")
-			checkPid := checkProcess(expecter)
-
-			Expect(checkPid).To(Equal(startPid))
+			expecter, _, err := console.NewExpecter(virtClient, vmi, 10*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+			defer expecter.Close()
 
+			By("Verifying VMI was indeed Paused")
+			uptimeDiffAfterPausing := hostUptime() - grepGuestUptime(expecter)
+			Expect(uptimeDiffAfterPausing).To(BeNumerically(">", uptimeDiffBeforePausing))
```

Comment:
I am actually surprised that  this can work. First it is probably only a matter of luck that the uptimeDiff is bigger on the second invocation here. Second this may probably only work for as long as virt-launcher did not yet sync the guest clock, because as far as I know, uptime shows uptime based on the wallclock compared to the last boot, so a pause difference should disappear as soon as the clock got synced by our agent.

Reply:
> I am actually surprised that this can work. First it is probably only a matter of luck that the uptimeDiff is bigger on the second invocation here. Second this may probably only work for as long as virt-launcher did not yet sync the guest clock, because as far as I know, uptime shows uptime based on the wallclock compared to the last boot, so a pause difference should disappear as soon as the clock got synced by our agent.

Thanks @rmohr  for explaining. I understand. In that case, to be frank, I don't know how to proceed, do you have any idea how to pull this off?

I added a PR for fixing the existing test in any case : #4420

Reply:
@rmohr  I check it manually and it worked for me. I am not sure if guest agent is installed by default on cirros( should appear in the [yaml](https://pastebin.pl/view/d035c5a9) doesn't it?). Anyway, I delayed the unpausing for 10 seconds and check the diff increased accordingly, to reduce the luck part in this test, and apparently it did work. could there be another reason that make it work?

Reply:
With cirros (since no guest agent present) it should work. :+1: